### PR TITLE
Add exclusion for users with private link

### LINKED ISSUE #46565: Add exclusion for users with private link
## Issue Description

Exclude users from the coming soon screen when they possess a private link.

## Acceptance criteria
* Exclude users with a valid private link (the `woo-share` param must match the `woocommerce_share_key` option value)
* Persist the `woo-share` param into a cookie that expires in 90 days (caches may be bypassed when the cookie value is present, so we don't want the cookie to live forever).

### PR DESCRIPTION
**Pull request by adrianduffell**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Excludes the coming soon screen from users with a private link.

Closes https://github.com/woocommerce/woocommerce/issues/46565.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. As an admin, enable sharing with a private link in Settings > Site visibility
2. copy the private link and open the URL in a new private window
3. See that the coming soon screen is bypassed
4. Navigate to other pages protected pages and see that the bypass continues to persist

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Internal/ComingSoon/ComingSoonRequestHandler.php [hunk 1]
```diff
@@ -83,4 +92,10 @@ private function render_404() {
 		}
 		die();
 	}
+
+	private function maybe_set_cookie() {
```
Comment (chihsuan) [inline plugins/woocommerce/src/Internal/ComingSoon/ComingSoonRequestHandler.php]: It seems that this method is unused. Should we remove it?

**Review by chihsuan — APPROVED** [review]
Thanks, tested well! 👍

File: plugins/woocommerce/src/Internal/ComingSoon/ComingSoonRequestHandler.php (hunk 1 above)
Comment (adrianduffell) [reply plugins/woocommerce/src/Internal/ComingSoon/ComingSoonRequestHandler.php]: Yup good idea, it was an idea I was exploring but we don't need it.  FIxed in 048dd1a

**Review by rjchow — COMMENTED** [review]
The cookie bit doesn't seem to work - It correctly saves the cookie but if I remove the woo-share query parameter (either manually or by navigating), the coming soon page comes up

**Comment by adrianduffell** [comment]
> The cookie bit doesn't seem to work - It correctly saves the cookie but if I remove the woo-share query parameter (either manually or by navigating), the coming soon page comes up

Thanks @rjchow , I think my lint fixes broke it by converting it to lowercase. I've corrected it in daf468f

**Review by rjchow — APPROVED** [review]
Working great now!

**Pull request merged**